src/main/scala/mesosphere/marathon/state/Volume.scala

```diff
@@ -213,20 +215,25 @@ object PersistentVolumeInfo {
       info.maxSize.forall(_ > info.size)
     }
 
+    val haveProperProfileName = validator[String] { profileName =>
+      profileName is notEmpty
+    }
+
     validator[PersistentVolumeInfo] { info =>
       info.size should be > 0L
       info.constraints.each must complyWithVolumeConstraintRules
       info should meetMaxSizeConstraint
       info should notHaveConstraintsOnRoot
       info should haveProperlyOrderedMaxSize
+      info.profileName is optional(haveProperProfileName)
```

Comment:
Nit: Is there a `should optionally(...)`?

Also, does this check make sense? `haveProperProfileName` validates that `profileName` is set but then the rule is optional. So it passes every time, doesn't it?

Reply:
AFAIU, `should` is a synonym of `is`. I am going to change it to `should` for consistency.

`optional` here means that if `profileName` is `Some(...)` then that validator gets applied to `...`, otherwise it results in `Success`. Does it clear it out?